### PR TITLE
Update documentation on guideline for test images to be added to/pulled from ghcr

### PR DESCRIPTION
**Pull request by lakshmimsft**
# Description

1. Updating documentation for test images to be pulled from ghcr and describing how to add new images to ghcr.
2. Including an instance of redis: latest to be updated to new location. [Question on package permissions to be resolved when adding new version to existing image]

## Type of change

- This pull request is a minor refactor, code cleanup, test improvement, or other maintenance task and doesn't change the functionality of Radius (issue link optional).

### CODE REVIEW

File: docs/contributing/contributing-code/contributing-code-tests/tests-images-pushtoghcr.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
+
+## Using Standard Images in Tests
+
+When writing tests for our applications, it's common to use standard images for resources like Redis, Mongo, and others. By default, specifying an image with `redis:latest` or similar tags will pull the image from Docker Hub. This behavior is due to Docker's default configuration, which automatically searches the Docker Hub registry for images when no other registry is specified. 
+
+However, to avoid potential rate limiting and network issues associated with pulling images from Docker Hub, we are transitioning to using images hosted in our public GitHub Container Registry (GHCR) repository. This change ensures more reliable access to the images needed for our tests and helps reduce potential disruptions caused by external dependencies.
+
+### New Image Pull Guideline
```
Comment (vinayada1) [inline docs/contributing/contributing-code/contributing-code-tests/tests-images-pushtoghcr.md:8]: I think we should just write this the way it is supposed to work now.....stay away from comparing old way to new.

Comment (lakshmimsft) [reply docs/contributing/contributing-code/contributing-code-tests/tests-images-pushtoghcr.md:8]: updated

File: docs/contributing/contributing-code/contributing-code-tests/tests-images-pushtoghcr.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+Moving forward, we will pull standard test images from our public GHCR repository instead of Docker Hub. 
+
+For example, instead of using:
+```yaml
+image: 'redis:latest'
+```
+We should now use:
+```yaml
+image: ghcr.io/radius-project/mirror/redis:latest
```
Comment (sk593) [inline docs/contributing/contributing-code/contributing-code-tests/tests-images-pushtoghcr.md:18]: What is the 'mirror' desciptor meant to be?

File: docs/contributing/contributing-code/contributing-code-tests/tests-images-pushtoghcr.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+```
+This change applies to all standard images previously pulled from Docker Hub. Please update any new tests accordingly.
+
+### Adding Images to GHCR Repository
+
```
Comment (sk593) [inline docs/contributing/contributing-code/contributing-code-tests/tests-images-pushtoghcr.md:23]: Can we add what roles you need to have to update these permissions? i.e. approvers can push to GHCR, maintainers can change package permissions (not sure if these are accurate but just examples)

File: docs/contributing/contributing-code/contributing-code-tests/tests-images-pushtoghcr.md (hunk 2 above)
Comment (lakshmimsft) [reply docs/contributing/contributing-code/contributing-code-tests/tests-images-pushtoghcr.md:18]: this was suggested by @youngbupark [here](https://github.com/radius-project/radius/pull/7530#discussion_r1577312961)

File: docs/contributing/contributing-code/contributing-code-tests/tests-images-pushtoghcr.md (hunk 3 above)
Comment (lakshmimsft) [reply docs/contributing/contributing-code/contributing-code-tests/tests-images-pushtoghcr.md:23]: In our meeting, it was decided that maintainers should have admin access while it's read only for all others. We can bring it up again if this needs to be reconsidered.

File: docs/contributing/contributing-code/contributing-code-tests/tests-images-pushtoghcr.md (hunk 2 above)
Comment (ytimocin) [reply docs/contributing/contributing-code/contributing-code-tests/tests-images-pushtoghcr.md:18]: Just checked one more time and we don't have Redis latest tag in our packages: https://github.com/orgs/radius-project/packages/container/package/mirror%2Fredis.

File: docs/contributing/contributing-code/contributing-code-tests/tests-images-pushtoghcr.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+When writing tests for our applications, it's common to use standard images for resources like Redis, Mongo, and others. By default, specifying an image with `redis:latest` or similar tags will pull the image from Docker Hub. This behavior is due to Docker's default configuration, which automatically searches the Docker Hub registry for images when no other registry is specified. 
+
+ We are transitioning to using images hosted in our public GitHub Container Registry (GHCR) repository. This change ensures more reliable access to the images needed for our tests and will help reduce potential issues caused by external dependencies.
+
+### New Image Pull Guideline
+
+Moving forward, we will pull standard test images from our public GHCR repository instead of Docker Hub. 
+
+For example, instead of using:
```
[the remark this answers is not in the document]
Comment (lakshmimsft) [reply docs/contributing/contributing-code/contributing-code-tests/tests-images-pushtoghcr.md]: updated

File: docs/contributing/contributing-code/contributing-code-tests/tests-images-pushtoghcr.md (hunk 2 above)
Comment (lakshmimsft) [reply docs/contributing/contributing-code/contributing-code-tests/tests-images-pushtoghcr.md:15]: All other tests updated for this refer to redis:6.2. This test works as expected with the label. Yes, we do not have the image with latest tag. if it is really needed for this test, I'll need to request rights from maintainers to add it to the image repo.

**Review by ytimocin — APPROVED** [review]

**Pull request merged**